src/material/paginator/paginator.ts

```diff
@@ -338,18 +347,24 @@ export abstract class _MatPaginatorBase<
 
     // If no page size is provided, use the first page size option or the default page size.
     if (!this.pageSize) {
-      this._pageSize =
+      const pageSize =
         this.pageSizeOptions.length != 0 ? this.pageSizeOptions[0] : DEFAULT_PAGE_SIZE;
+      this._pageSize = typeof pageSize === 'number' ? pageSize : pageSize.value;
     }
 
-    this._displayedPageSizeOptions = this.pageSizeOptions.slice();
+    this._displayedPageSizeOptions = this.pageSizeOptions.map(pageSize => {
+      if (typeof pageSize === 'number') {
+        return {value: pageSize, label: `${pageSize}`};
+      }
+      return pageSize;
+    });
 
-    if (this._displayedPageSizeOptions.indexOf(this.pageSize) === -1) {
-      this._displayedPageSizeOptions.push(this.pageSize);
+    if (this._displayedPageSizeOptions.map(x => x.value).indexOf(this.pageSize) === -1) {
```

Comment:
Instead of using `map` and `indexOf`, you can use `Array.prototype.some`.

Reply:
done